### PR TITLE
Moved jiffies to hal. Use hal_getTimer() for both systick and hptimer…

### PR DESCRIPTION
**Pull request by agkaminski**
… based systems

JIRA: RTOS-79

Use hal_getTimer() as a getter for jiffies. Needed to be able to provide source other than systick as a general system timer.

## Motivation and Context

Needed to be able to provide source other than systick as a general system timer.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (stm32l4x6).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by Maxez — DISMISSED** [review]
LGTM

File: hal/armv7m/imxrt/timer.c [hunk 1]
```diff
@@ -24,7 +25,47 @@
...
+
+
+int _timer_irqHandler(unsigned int n, cpu_context_t *ctx, void *arg)
+{
+	(void)n;
+	(void)arg;
+	(void)ctx;
+
+	timer_common.jiffies += timer_common.interval;
+	return -1;
```
Comment (HBuczynski) [inline hal/armv7m/imxrt/timer.c:44]: Shoudn't be `0`?

File: hal/armv7m/stm32/l1/timer.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+
+int _timer_irqHandler(unsigned int n, cpu_context_t *ctx, void *arg)
+{
+	(void)n;
+	(void)arg;
+	(void)ctx;
+
+	timer_common.jiffies += timer_common.interval;
+	return -1;
```
Comment (HBuczynski) [inline hal/armv7m/stm32/l1/timer.c:37]: Shoudn't be `0`?

File: hal/ia32/timer.c [hunk 3]
```diff
@@ -16,15 +16,44 @@
...
 
 
+static int timer_irqHandler(unsigned int n, cpu_context_t *ctx, void *arg)
+{
+	(void)n;
+	(void)arg;
+	(void)ctx;
+
+	timer.jiffies += timer.interval;
+	return -1;
```
Comment (HBuczynski) [inline hal/ia32/timer.c:40]: Shoudn't be `0`?

File: hal/riscv64/timer.c [hunk 4]
```diff
@@ -16,15 +16,44 @@
...
 
 
+static int timer_irqHandler(unsigned int n, cpu_context_t *ctx, void *arg)
+{
+	(void)n;
+	(void)arg;
+	(void)ctx;
+
+	timer.jiffies += timer.interval;
+	return -1;
```
Comment (HBuczynski) [inline hal/riscv64/timer.c:40]: Shoudn't be `0`?

File: hal/armv7m/timer.h [hunk 5]
```diff
@@ -24,6 +24,12 @@
 #define TIMER_CYC2US(x) (x)
 
 
+extern void timer_jiffiesAdd(time_t t);
```
Comment (HBuczynski) [inline hal/armv7m/timer.h:27]: I know that this function is not used on `imxrt` but this is a part of the interface.  I think it should be implemented as well.

File: hal/armv7m/imxrt/timer.c [hunk 6]
```diff
@@ -24,7 +25,47 @@
...
+
+	hal_spinlockSet(&timer_common.sp, &sc);
+	ret = timer_common.jiffies;
+	hal_spinlockClear(&timer_common.sp, &sc);
+
+	return ret;
+}
+
+
 void _timer_init(u32 interval)
```
Comment (HBuczynski) [inline hal/armv7m/imxrt/timer.c:61]: `timer_common.jiffies` is not initialized with value 0. I am not sure whether `.bss` is zeored in kernel. This applies to all platforms.

File: proc/threads.c [hunk 7]
```diff
@@ -432,14 +431,7 @@ static void _threads_updateWakeup(time_t now, thread_t *min)
 
 static inline time_t _threads_getTimer(void)
```
Comment (HBuczynski) [inline proc/threads.c:432]: After your changes, `_threads_getTimer` invokes only `hal_getTimer` and doesn't do anything else. 
What do you think about removing `_threads_getTimer` and using only `hal_getTimer`?

**Review by HBuczynski — CHANGES_REQUESTED** [review]

File: hal/armv7m/imxrt/timer.c (hunk 1 above)
Comment (agkaminski) [reply hal/armv7m/imxrt/timer.c]: Done

File: hal/ia32/timer.c (hunk 3 above)
Comment (agkaminski) [reply hal/ia32/timer.c]: Done

File: hal/armv7m/stm32/l1/timer.c (hunk 2 above)
Comment (agkaminski) [reply hal/armv7m/stm32/l1/timer.c]: Done

File: hal/riscv64/timer.c (hunk 4 above)
Comment (agkaminski) [reply hal/riscv64/timer.c]: Done

File: hal/armv7m/timer.h (hunk 5 above)
Comment (agkaminski) [reply hal/armv7m/timer.h:27]: Done

File: hal/armv7m/imxrt/timer.c (hunk 6 above)
Comment (agkaminski) [reply hal/armv7m/imxrt/timer.c:67]: Good find, it's not zeroed indeed

File: proc/threads.c (hunk 7 above)
Comment (agkaminski) [reply proc/threads.c]: I was planning on doing that in the next commits, but can change it now as well

**Review by HBuczynski — APPROVED** [review]
I have also tested on `imxrt106x` and it works fine.
LGTM

**Pull request merged**